x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/RolloverStep.java

```diff
@@ -39,38 +40,46 @@ public boolean isRetryable() {
     @Override
     public void performAction(IndexMetadata indexMetadata, ClusterState currentClusterState,
                               ClusterStateObserver observer, Listener listener) {
-        boolean indexingComplete = LifecycleSettings.LIFECYCLE_INDEXING_COMPLETE_SETTING.get(indexMetadata.getSettings());
-        if (indexingComplete) {
-            logger.trace(indexMetadata.getIndex() + " has lifecycle complete set, skipping " + RolloverStep.NAME);
-            listener.onResponse(true);
-            return;
-        }
+        IndexAbstraction indexAbstraction = currentClusterState.metadata().getIndicesLookup().get(indexMetadata.getIndex().getName());
+        final String rolloverTarget;
+        if (indexAbstraction.getParentDataStream() != null) {
+            rolloverTarget = indexAbstraction.getParentDataStream().getDataStream().getName();
```

Comment:
Just to be super paranoid, I think we should handle the case where `getDataStream()` returns null

Reply:
I think that can be changed to:

```
rolloverTarget = indexAbstraction.getParentDataStream().getName();
```

to eliminate the need for another null check.